dripconfig/builtins.py

```diff
@@ -82,16 +83,33 @@ class StatsdConfig(SchemaBasedTrigger):
         'statsd': {
             Required('host'): basestring,
             Required('port', default=8125): Coerce(int),
+            Optional('sample_rate'): Coerce(int),
+            Optional('disabled'): bool,
         },
     })
 
     def configure(self, configuration):
+        """
+        If statsd is missing from the config, the emission of
+        data will be disabled by default. Otherwise, use the provided defaults,
+        or use the values specified in the config.
+        """
         import statsd
 
-        if 'statsd' in configuration:
-            statsd.Connection.set_defaults(
-                host=configuration.statsd.host,
-                port=configuration.statsd.port)
+        statsd_in_conf = 'statsd' in configuration
+        defaults = {
```

Comment:
why set defaults here and not in the schema, which has a whole facility for setting defaults?